gobblin-core/src/main/java/org/apache/gobblin/qualitychecker/row/RowLevelPolicyChecker.java

```diff
@@ -197,7 +198,23 @@ public State getFinalState() {
    * @return a {@link ControlMessageHandler}.
    */
   protected ControlMessageHandler getMessageHandler() {
-    return ControlMessageHandler.NOOP;
+    /**
+     * When seeing {@link FlushControlMessage and using ERR_FILE as the quality-checker handling,
+     * close the open error file and create new one.
+     */
+    return new ControlMessageHandler() {
+      @Override
+      public void handleMessage(ControlMessage message) {
+        if (message instanceof FlushControlMessage ) {
+          try {
+            RowLevelPolicyChecker.this.close();
+            RowLevelPolicyChecker.this.writer = new RowLevelErrFileWriter(RowLevelPolicyChecker.this.fs);
```

Comment:
Can the creation be moved inside checkResult() where we check if the file is open or not?